synapse/handlers/typing.py

```diff
@@ -362,11 +362,7 @@ async def _recv_edu(self, origin: str, content: JsonDict) -> None:
             )
             return
 
-        domains = await self._storage_controllers.state.get_current_hosts_in_room(
-            room_id
-        )
-
-        if self.server_name in domains:
+        if await self.store.is_host_joined(room_id, self.server_name):
```

Comment:
Can we add a comment to explain why we use `is_host_joined` instead of `get_current_hosts_in_room`?

```suggestion
        # Check whether the origin server is in the room.
        # For rooms with partial state, we ignore most typing EDUs until fully synced.
        if await self.store.is_host_joined(room_id, self.server_name):
```

Reply:
I don't think we ignore it here ?
I think we use the servers list we get on the beginning of the fast join to decide to process or not, and the typing event even seems to come down on the next sync after fully joining the room according to the complement test.

Reply:
I'm fairly we sure we don't use the server list. It's stored in `partial_state_rooms_servers` and the only thing that reads from that table is the code to restart the sync of full state.

Reply:
Right so `get_current_hosts_in_room` would wait for the server list, while `is_host_joined` is looking at the room memberships, and we have the joining event creating when launching the fast join, so success.
I am going to avoid saying we ignore the typing EDUs however because it doesn't seem to be the case, it does get delivered in sync.

Reply:
It looks like `self.server_name` is actually our own homeserver, and the check will always pass?
How the previous code worked is a mystery to me.

Reply:
Hum yes I just noticed that a similar check is done on L343... It seems like the initial code is wrong indeed.

It doesn't seem like we do a check further down either, I can't even spot one in the sync code after a quick glance.

Well I guess it's time to check that with a test and fix it if needed.

Reply:
Actually we do the check of the user id against the origin of the fed request right before:
https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/typing.py#L359-L363

Looks like this `if` is just redundant.